### DISCUSSION TITLE
Change scope for InitializeFromData() in page Item Availability Check

### DISCUSSION THREAD

**Discussion by sadjad1** [microsoft/ALAppExtensions#19847] [Ideas] [answered]
Hi!
I have come across clients who **explicitly** wants to be able to trigger _page 1872 "Item Availability Check"_ manually for specifick Sales Order Lines. There of course other ways around to show the same data as the in this page _1872 "Item Availability Check"_ does but they explicitly asks for this page.

So I was checking this page out and see what's possible and not. What I could find was a function in this page called InitializeFromData. This function could be useful in order to manually trigger this page but is that has the scope [Scope('OnPrem')]. Is it possible to remove this scope for Bussiness Central that is **not** on premesis so we can use this function @JesperSchulz?

![image](https://user-images.githubusercontent.com/15630953/184585378-23afc1ab-a524-44b9-8eb5-c3d451ce526c.png)

**Comment by JesperSchulz** [comment, marked as answer]
I do not see why that function must have scope "OnPrem". Please create an issue with this extensibility request here on GitHub, and we will take a closer look :-)

**Reply by sadjad1** [reply]
Done and thank you for quick answer :)
Issue: https://github.com/microsoft/ALAppExtensions/issues/19930
